src/megatron/bridge/data/vlm_datasets/hf_provider.py

```diff
@@ -0,0 +1,116 @@
+# Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Provider that builds conversation datasets from HuggingFace datasets.
+"""
+
+from dataclasses import dataclass
+from typing import Any, Callable, Dict, List, Optional, Tuple
+
+import torch
+from transformers import AutoProcessor
+
+from megatron.bridge.training.config import DatasetBuildContext, DatasetProvider
+
+from .conversation_dataset import VLMConversationDataset
+from .hf_dataset_makers import (
+    make_cord_v2_dataset,
+    make_cv17_dataset,
+    make_medpix_dataset,
+    make_rdr_dataset,
+)
+
+
+@dataclass(kw_only=True)
+class HFDatasetConversationProvider(DatasetProvider):
```

Comment:
did you have any issues when not specifying the dataloader type, which controls the samplers attached to the dataloader?